### PR TITLE
gh-105481: remove HAS_ARG, HAS_CONST, IS_JUMP_OPCODE, IS_PSEUDO_OPCODE and replace by their new versions

### PR DESCRIPTION
**Pull request by iritkatriel**
* Issue: gh-105481

### CODE REVIEW

**Review by gvanrossum — APPROVED** [review]
LGTM. Do we need any doc update for `dis.stack_effect` (or is it `opcode.stack_effect` now that it no longer requires the presence of oparg to match the requirements of the opcode.

**Comment by iritkatriel** [comment]
The doc doesn't mention the arg check issue. Should I add a versionchanged note anyway?

**Comment by gvanrossum** [comment]
Might as well.

File: Misc/NEWS.d/next/Library/2023-06-17-12-13-57.gh-issue-105481.KgBH5w.rst [hunk 1]
```diff
@@ -0,0 +1,4 @@
+:func:`~dis.stack_effect`B no longer raises an exception if an ``oparg`` is
```
Comment (arhadthedev) [inline Misc/NEWS.d/next/Library/2023-06-17-12-13-57.gh-issue-105481.KgBH5w.rst]:
```suggestion
:func:`~dis.stack_effect` no longer raises an exception if an ``oparg`` is
```

Comment (arhadthedev) [reply Misc/NEWS.d/next/Library/2023-06-17-12-13-57.gh-issue-105481.KgBH5w.rst]: There was a race between me and the commit, thus marking as resolved.

**Review by gvanrossum — APPROVED** [review]

**Pull request merged**